### ISSUE TITLE
WeChat QR Code

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin-dollar#112]
Our footer includes an icon for WeChat. The link that it's pointing to is derived from a QR code. It doesn't actually work when visited directly in a browser or scanned from outside of the WeChat app. So instead of linking to this URL, we should display the QR code when people click the icon. I recommend opening a modal that contains the QR code along with a simple instruction. I prefer this over a QR code floating near the icon when someone triggers the hover event.

**Comment by topoface** [comment]
@micahalcorn Is this issue still valuable?

**Comment by topoface** [comment]
I am not sure if I misunderstood the needs, but please tell me if so.

**Comment by micahalcorn** [comment]
@topoface yeah, this would be nice to have. Currently, the footer at [ousd.com](https://ousd.com) has a WeChat icon linking to https://u.wechat.com/MPuvZYXixbeMimCix6GzRcw. If you click that in a browser, it just redirects to the WeChat hope page. Instead, we should give users a QR code code that can be scanned from the app like we do on https://www.originprotocol.com/ ***except*** I was suggesting that it be a proper modal with instructions to scan with the app rather than showing just the QR on hover like we do there. I'm not entirely sure how to handle this on mobile screens since they can't scan themselves and linking to the URL doesn't seem to work.

**Comment by topoface** [comment]
Oh, I understand.
I was thinking it was for the main website.

**Comment by topoface** [comment]
@micahalcorn 
We can display modal or public image as well, but in case of mobile, the problem can't be solved.

So, here's my thought.
1. Create a public url of the image
2. If the user click the wechat icon, then the image is opened in new tab.
3. In case of mobile, user can download that image and scan it from file.

**Comment by micahalcorn** [comment]
> @micahalcorn
> We can display modal or public image as well, but in case of mobile, the problem can't be solved.
> 
> So, here's my thought.
> 
> 1. Create a public url of the image
> 2. If the user click the wechat icon, then the image is opened in new tab.
> 3. In case of mobile, user can download that image and scan it from file.

Ah, thanks, I didn't realize that you could choose from library in the WeChat app. 🤦 Do we actually need to host the image or can we just generate it dynamically in the modal?

**Comment by topoface** [comment]
They are similar in how they work.
But I think hosting image is pretty simple.

**Comment by micahalcorn** [comment]
> They are similar in how they work.
> But I think hosting image is pretty simple.

Sounds good 👍 Please let me know what else you need from us in regards to content or design and let's make it happen.

**Comment by topoface** [comment]
@micahalcorn 
I am not sure if you're using AWS, but in that case, the image may be on the S3 bucket I think.
